package/expo-package/src/optionalDependencies/takePhoto.ts

```diff
@@ -36,44 +43,63 @@ export const takePhoto = ImagePicker
 
         if (permissionGranted) {
           const imagePickerSuccessResult = await ImagePicker.launchCameraAsync({
+            mediaTypes: mediaTypeMap[mediaType],
             quality: Math.min(Math.max(0, compressImageQuality), 1),
           });
           const canceled = imagePickerSuccessResult.canceled;
           const assets = imagePickerSuccessResult.assets;
           // since we only support single photo upload for now we will only be focusing on 0'th element.
           const photo = assets && assets[0];
-
-          if (canceled === false && photo && photo.height && photo.width && photo.uri) {
-            let size: Size = {};
-            if (Platform.OS === 'android') {
-              const getSize = (): Promise<Size> =>
-                new Promise((resolve) => {
-                  Image.getSize(photo.uri, (width, height) => {
-                    resolve({ height, width });
-                  });
-                });
-
-              try {
-                const { height, width } = await getSize();
-                size.height = height;
-                size.width = width;
-              } catch (e) {
-                console.warn('Error get image size of picture caputred from camera ', e);
-              }
-            } else {
-              size = {
-                height: photo.height,
-                width: photo.width,
-              };
-            }
-
+          if (canceled) {
+            return { cancelled: true };
+          }
+          if (photo.mimeType.includes('video')) {
+            const clearFilter = new RegExp('[.:]', 'g');
+            const date = new Date().toISOString().replace(clearFilter, '_');
             return {
+              ...photo,
               cancelled: false,
-              size: photo.fileSize,
+              duration: photo.duration, // in milliseconds
               source: 'camera',
+              name: 'video_recording_' + date + photo.uri.split('.').pop(),
+              size: photo.fileSize,
+              type: photo.mimeType,
               uri: photo.uri,
-              ...size,
             };
+          } else {
```

Comment:
I assume the `mediaTypes` from above ensure that we would only get those ? If not, perhaps we should also check if `photo.mimeType.includes('image')` here too ? I know we didn't before, but if there could be other mime types I would prefer to be explicit. If other mime types are not viable then feel free to disregard.

Reply:
I think it only returns image and video so we are good to go.